static/app/views/replays/replayTable/index.tsx

```diff
@@ -36,161 +39,190 @@ type Props = {
   visibleColumns: ReplayColumn[];
   emptyMessage?: ReactNode;
   gridRows?: string;
+  onClickPlay?: (index: number) => void;
+  replayPlayButtonPriority?: React.ComponentProps<typeof Button>['priority'];
+  selectedReplayIndex?: number;
   showDropdownFilters?: boolean;
 };
 
-function ReplayTable({
-  fetchError,
-  isFetching,
-  replays,
-  sort,
-  visibleColumns,
-  emptyMessage,
-  gridRows,
-  showDropdownFilters,
-}: Props) {
-  const routes = useRoutes();
-  const location = useLocation();
-  const organization = useOrganization();
-
-  const tableHeaders = visibleColumns
-    .filter(Boolean)
-    .map(column => <HeaderCell key={column} column={column} sort={sort} />);
-
-  if (fetchError && !isFetching) {
+// Memoizing this component to avoid unnecessary re-renders
+const ReplayTable = memo(
```

Comment:
This is sus, we might be able to avoid this by a) not using `{...props}` if we can, and moving the state that these buttons care about into the url. This way fewer props have to pass through the whole table we just want the one or two cells to update.

I would avoid using spread operations as much as possible. If we need to pass, like idk 10 props, to a component then that interface is not great... but we shouldn't hide it with a spread that potentially includes more values and causes re-renders

Reply:
@ryan953 The problem is the parent component re-renders every time the time changes. I need the replay context as a parent so I can set the button state.